airbyte-cdk/python/airbyte_cdk/utils/schema_inferrer.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Dict, List, Optional, Union
+
+from airbyte_cdk.models import AirbyteRecordMessage
+from genson import SchemaBuilder
+from genson.schema.strategies.object import Object
+
+
+class NoRequiredObj(Object):
+    """
+    This class has Object behaviour, but it does not generate "required[]" fields
+    every time it parses object. So we dont add unnecessary extra field.
+    """
+
+    def to_schema(self):
+        schema = super(Object, self).to_schema()
+        schema["type"] = "object"
+        if self._properties:
```

Comment:
are these two if blocks what prevent generating `required` fields?

Reply:
In a sense, I copied this over from https://github.com/airbytehq/airbyte/blob/master/tools/schema_generator/schema_generator/infer_schemas.py#L41

However on checking the genson code I changed the logic to just call the existing `to_schema` logic of the object strategy and then dropping the `required` prop. It's doing a very small amount of additional work but it seems much cleaner, LMK what you think.